coderd/database/queries/auditlogs.sql

```diff
@@ -14,6 +14,26 @@ FROM
 	audit_logs
 LEFT JOIN
     users ON audit_logs.user_id = users.id
+LEFT JOIN
+		-- First join on workspaces to get the initial workspace create
+		-- to workspace build 1 id. This is because the first create is
+		-- is a different audit log than subsequent starts.
+    workspaces on
+                audit_logs.resource_type = 'workspace'
+            AND audit_logs.resource_id = workspaces.id
+    LEFT JOIN
+            workspace_builds on
+                -- Get the reason from the build if the resource type
+                -- is a workspace_build
+                (audit_logs.resource_type = 'workspace_build'
+                    AND audit_logs.resource_id = workspace_builds.id)
+                OR
+                -- Get the reason from the build #1 if this is the first
+                -- workspace create.
+                (audit_logs.resource_type = 'workspace' AND
+                audit_logs.action = 'create' AND
+                 workspaces.id = workspace_builds.workspace_id AND
+                 workspace_builds.build_number = 1)
```

Comment:
The above is a bit obnoxious, but it exists to handle the case when a workspace is first created. In these situations, the `resource_type` is `workspace`, not `workspace_build`.

Otherwise, we could simplify: 
`LEFT JOIN
    workspace_builds on audit_logs.resource_type = ‘workspace_build’ AND audit_logs.resource_id = workspace_builds.id`

Reply:
I think we need to fix the formatting on this sql.

:+1:

Reply:
@Emyrk I think I fixed it with `pgFormatter` but please let me know if it still looks wonky.

Reply:
I don't think it did, but we don't abide by some standard sql formatting because the `sqlc` directives are not valid sql, and the formatters do not know what to do :shrug:.

```sql
-- GetAuditLogsBefore retrieves `row_limit` number of audit logs before the provided
-- ID.
-- name: GetAuditLogsOffset :many
SELECT
	audit_logs.*,
	users.username AS user_username,
	users.email AS user_email,
	users.created_at AS user_created_at,
	users.status AS user_status,
	users.rbac_roles AS user_roles,
	users.avatar_url AS user_avatar_url,
	COUNT(audit_logs.*) OVER() AS count
FROM
	audit_logs
	LEFT JOIN
		users ON audit_logs.user_id = users.id
	LEFT JOIN
		-- First join on workspaces to get the initial workspace create
		-- to workspace build 1 id. This is because the first create is
		-- is a different audit log than subsequent starts.
		workspaces on
			audit_logs.resource_type = 'workspace'
			AND audit_logs.resource_id = workspaces.id
	LEFT JOIN
		workspace_builds on
			-- Get the reason from the build if the resource type
			-- is a workspace_build
			(
			    audit_logs.resource_type = 'workspace_build' AND
			    audit_logs.resource_id = workspace_builds.id
			)
			OR
			-- Get the reason from the build #1 if this is the first
			-- workspace create.
			(
			    audit_logs.resource_type = 'workspace' AND
				audit_logs.action = 'create' AND
				workspaces.id = workspace_builds.workspace_id AND
				workspace_builds.build_number = 1
			)
-- WHERE clause stuff below. I think the indentation is messed up there too a bit.
```